test/linux/example/media_on_linux/SocketCANInterfaceManager.cpp

```diff
@@ -0,0 +1,285 @@
+/*
```

Comment:
Because this is an example. It's easier to follow code when it has a clean interface (header) and the implementation is separate. Header-only for the core library is a concession to the limitations of C++ and templates. It is not a generally desirable pattern.